python/ray/autoscaler/_private/aws/cloudwatch/cloudwatch_helper.py

```diff
@@ -59,33 +82,88 @@ def _ec2_health_check_waiter(self, node_ids: List[str]) -> None:
+                                  is_head_node: bool) -> None:
+        """
+        check whether update operations are needed in
+        cloudwatch related configs
         """
         cwa_installed = self._setup_cwa()
-        param_name = self._get_ssm_param_name()
+        param_name = self._get_ssm_param_name(config_type)
         if cwa_installed:
             if is_head_node:
                 cw_config_ssm = self._set_cloudwatch_ssm_config_param(
-                    param_name)
-                cur_cw_config_hash = self._sha1_hash_file()
+                    param_name, config_type)
+                cur_cw_config_hash = self._sha1_hash_file(config_type)
                 ssm_cw_config_hash = self._sha1_hash_json(cw_config_ssm)
-                # check if user updated Unified Cloudwatch Agent config file.
+                # check if user updated cloudwatch related config files.
                 # if so, perform corresponding actions.
                 if cur_cw_config_hash != ssm_cw_config_hash:
                     logger.info(
-                        "Unified Cloudwatch Agent config file has changed.")
-                    self._upload_config_to_ssm_and_set_hash_tag()
-                    self._restart_cloudwatch_agent()
+                        "Cloudwatch {} config file has changed.".format(
+                            config_type))
+                    self._upload_config_to_ssm_and_set_hash_tag(config_type)
+                    self.CLOUDWATCH_CONFIG_TYPE_TO_UPDATE_FUNC.get(
+                        config_type)()
             else:
-                head_node_hash = self._get_head_node_config_hash()
-                cur_node_hash = self._get_cur_node_config_hash()
+                head_node_hash = self._get_head_node_config_hash(config_type)
+                cur_node_hash = self._get_cur_node_config_hash(config_type)
                 if head_node_hash != cur_node_hash:
                     logger.info(
-                        "Unified Cloudwatch Agent config file has changed.")
-                    self._restart_cloudwatch_agent()
+                        "Cloudwatch {} config file has changed.".format(
+                            config_type))
+                    self.CLOUDWATCH_CONFIG_TYPE_TO_UPDATE_FUNC.get(
+                        config_type)()
                     self._update_cloudwatch_hash_tag_value(
-                        self.node_ids, head_node_hash)
+                        self.node_ids, head_node_hash, config_type)
+
+    def _put_cloudwatch_dashboard(self) -> Dict[str, Any]:
+        """put dashboard to cloudwatch console"""
+
+        cloudwatch_config = self.provider_config["cloudwatch"]
+        dashboard_config = cloudwatch_config \
+            .get(CloudwatchConfigType.DASHBOARD.value, {})
+        dashboard_name_cluster = dashboard_config.get("name",
+                                                      self.cluster_name)
+        dashboard_name = dashboard_name_cluster + "-" + "".join(self.node_ids)
```

Comment:
If we're expecting self.node_ids to just be one cluster node here, then we should validate this explicitly. Also, I'm not totally sure how I feel about having a dashboard per cluster instance, since a design that presents a cluster-level dashboard seems more useful.